src/hooks/fetchUtils.test.ts

```diff
@@ -0,0 +1,229 @@
+            headers: { 'content-type': 'application/json' },
+            body: JSON.stringify({ data: '12345' }),
+          },
+        },
+      ],
+      // Here we will add the init object, making sure it overrides the input values
+      [
+        {
+          // @ts-ignore
+          input: new Request('https://example.com', {
+            method: 'POST',
+            headers: { 'content-type': 'application/json' },
+            body: JSON.stringify({ data: '12345' }),
+          }),
+          init: {
+            method: 'PUT',
+            headers: { 'content-type': 'application/xml' },
+            body: JSON.stringify({ data: '54321' }),
+          },
+          expectedUrl: 'https://example.com/',
+          expectedOptions: {
+            method: 'PUT',
+            headers: { 'content-type': 'application/xml' },
+            body: JSON.stringify({ data: '54321' }),
+          },
+        },
+      ],
+      [
+        {
+          input: 'https://example.com',
+          init: {
+            method: 'PUT',
+            headers: { 'content-type': 'application/xml' },
+            body: JSON.stringify({ data: '54321' }),
+          },
+          expectedUrl: 'https://example.com',
+          expectedOptions: {
+            method: 'PUT',
+            headers: { 'content-type': 'application/xml' },
+            body: JSON.stringify({ data: '54321' }),
+          },
+        },
+      ],
+      [
+        {
+          input: new URL('https://example.com'),
+          init: {
+            method: 'PUT',
+            headers: { 'content-type': 'application/xml' },
+            body: JSON.stringify({ data: '54321' }),
+          },
+          expectedUrl: 'https://example.com/',
+          expectedOptions: {
+            method: 'PUT',
+            headers: { 'content-type': 'application/xml' },
+            body: JSON.stringify({ data: '54321' }),
+          },
+        },
+      ],
+      // Test different formats for body
+      [
+        {
+          input: 'https://example.com',
+          init: {
+            method: 'PUT',
+            headers: { 'content-type': 'application/xml' },
+            // @ts-ignore
+            body: new Blob(['Blob contents']),
+          },
+          expectedUrl: 'https://example.com',
+          expectedOptions: {
+            method: 'PUT',
+            headers: { 'content-type': 'application/xml' },
+            body: 'Blob contents',
+          },
+        },
+      ],
+      [
+        {
+          input: 'https://example.com',
+          init: {
+            method: 'PUT',
+            headers: { 'content-type': 'application/xml' },
+            // Unsupported body type
+            body: 123,
+          },
+          expectedUrl: 'https://example.com',
+          expectedOptions: {
+            method: 'PUT',
+            headers: { 'content-type': 'application/xml' },
+          },
+        },
+      ],
+      // TODO: Test FormData body
+      // TODO: Test ReadableStream body
```

Comment:
Are these planned to be added in this PR, or just a leftover comment?

Reply:
@harelmo-lumigo Leftover, I kept them while moving tests